renderdoc/os/win32/win32_shellext.cpp

```diff
@@ -121,6 +124,39 @@ struct RDCThumbnailProvider : public IThumbnailProvider, IInitializeWithStream
 
     delete[] buf;
 
+    if(is_dds_file(captureHeader.data(), numRead))
+    {
+      // Read File from beginning. Seek requires the struct LARGE_INTEGER.
+      LARGE_INTEGER zero;
+      zero.QuadPart = 0;
+      pstream->Seek(zero, STREAM_SEEK_SET, NULL);
+      STATSTG stats = {};
+      pstream->Stat(&stats, STATFLAG_DEFAULT);
+      ULARGE_INTEGER size = stats.cbSize;
+
+      byte *ddsFile = new byte[size.QuadPart];
+      ULONG ddsNumRead = 0;
+      pstream->Read(ddsFile, static_cast<ULONG>(size.QuadPart), &ddsNumRead);
```

Comment:
Please use normal `(ULONG)` casts instead of `static_cast`s.

Reply:
Is replaced now.